Errors with Praat 6

System calls to `praat` fail with Praat 6, as its command syntax has changed. But adding the newly required `--run` argument causes an error with Praat 5.

Sadly, the fix failed to solve the issue for the *legacy* plugin under *linux* with praat *6*:
```
:legacy-plugin:test

Gradle suite > Gradle test > de.dfki.mary.voicebuilding.BuildLogicFunctionalTest.testLegacyPraatPitchmarker FAILED
    org.gradle.testkit.runner.UnexpectedBuildFailure: Unexpected build execution failure in /home/psibre/workspace/gradle-marytts-voicebuilding-plugin/legacy-plugin/build/testProject with arguments [legacyPraatPitchmarker]

    Output:
    :processDataResources UP-TO-DATE
    :lab UP-TO-DATE
    :templates UP-TO-DATE
    :text UP-TO-DATE
    :wav UP-TO-DATE
    :legacyInit UP-TO-DATE
    :legacyPraatPitchmarker
    /home/psibre/praat6015/praat

    Loading database layout:
    Reading config file /home/psibre/workspace/gradle-marytts-voicebuilding-plugin/legacy-plugin/build/testProject/build/database.config
    Loading basename list from wav files
    Found 24 files in basename list
    Loading features from file /home/psibre/workspace/gradle-marytts-voicebuilding-plugin/legacy-plugin/build/testProject/build/mary/features.txt
    Running PraatPitchmarker
    Computing pitchmarks for 24 utterances.
    Running Praat as: /home/psibre/praat6015/praat /home/psibre/workspace/gradle-marytts-voicebuilding-plugin/legacy-plugin/build/testProject/build/temp/script.praat /home/psibre/workspace/gradle-marytts-voicebuilding-plugin/legacy-plugin/build/testProject/build/basenames.lst /home/psibre/workspace/gradle-marytts-voicebuilding-plugin/legacy-plugin/build/testProject/build/wav/ /home/psibre/workspace/gradle-marytts-voicebuilding-plugin/legacy-plugin/build/testProject/build/pm/ 75 300

    (praat:17580): GLib-GObject-WARNING **: invalid (NULL) pointer instance

    (praat:17580): GLib-GObject-CRITICAL **: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

    (praat:17580): Gtk-WARNING **: Screen for GtkWindow not set; you must always set
    a screen for a GtkWindow before using the window

    (praat:17580): Gdk-CRITICAL **: IA__gdk_screen_get_default_colormap: assertion 'GDK_IS_SCREEN (screen)' failed

    (praat:17580): Gdk-CRITICAL **: IA__gdk_colormap_get_visual: assertion 'GDK_IS_COLORMAP (colormap)' failed

    (praat:17580): Gdk-CRITICAL **: IA__gdk_screen_get_default_colormap: assertion 'GDK_IS_SCREEN (screen)' failed

    (praat:17580): Gdk-CRITICAL **: IA__gdk_screen_get_root_window: assertion 'GDK_IS_SCREEN (screen)' failed

    (praat:17580): Gdk-CRITICAL **: IA__gdk_screen_get_root_window: assertion 'GDK_IS_SCREEN (screen)' failed

    (praat:17580): Gdk-CRITICAL **: IA__gdk_window_new: assertion 'GDK_IS_WINDOW (parent)' failed
    Exception in thread "main" java.lang.RuntimeException: PraatPitchmarker computation failed on file []!
    Command line was: [/home/psibre/praat6015/praat /home/psibre/workspace/gradle-marytts-voicebuilding-plugin/legacy-plugin/build/testProject/build/temp/script.praat /home/psibre/workspace/gradle-marytts-voicebuilding-plugin/legacy-plugin/build/testProject/build/basenames.lst /home/psibre/workspace/gradle-marytts-voicebuilding-plugin/legacy-plugin/build/testProject/build/wav/ /home/psibre/workspace/gradle-marytts-voicebuilding-plugin/legacy-plugin/build/testProject/build/pm/ 75 300].
        at marytts.util.io.General.launchProc(General.java:549)
        at marytts.tools.voiceimport.PraatPitchmarker.compute(PraatPitchmarker.java:294)
        at marytts.tools.voiceimport.DatabaseImportMain.main(DatabaseImportMain.java:428)
    :legacyPraatPitchmarker FAILED

    FAILURE: Build failed with an exception.
```

--- a/data-plugin/src/main/groovy/de/dfki/mary/voicebuilding/tasks/PraatPitchmarkerTask.groovy
+++ b/data-plugin/src/main/groovy/de/dfki/mary/voicebuilding/tasks/PraatPitchmarkerTask.groovy
@@ -40,7 +40,7 @@ class PraatPitchmarkerTask extends DefaultTask {
         // test for Praat 6
         def praat = ['praat']
         def praatTestProc = [praat.first(), '--version'].execute()
-        praatTestProc.waitFor()
+        praatTestProc.waitForOrKill(100)
         if (praatTestProc.exitValue()) {
             // assume we have Praat 5
         } else {

--- a/legacy-plugin/src/main/groovy/de/dfki/mary/voicebuilding/VoicebuildingLegacyPlugin.groovy
+++ b/legacy-plugin/src/main/groovy/de/dfki/mary/voicebuilding/VoicebuildingLegacyPlugin.groovy
@@ -34,20 +34,6 @@ class VoicebuildingLegacyPlugin implements Plugin<Project> {
             def proc = 'which ch_track'.execute()
             proc.waitFor()
             speechToolsDir = new File(proc.in.text)?.parentFile?.parent
-
-            // configure praat
-            proc = 'which praat'.execute()
-            proc.waitFor()
-            praat = proc.in.text
-            // test for Praat 6
-            def praat = 'praat'
-            def praatTestProc = [praat, '--version'].execute()
-            praatTestProc.waitFor()
-            if (praatTestProc.exitValue()) {
-                // assume we have Praat 5
-            } else {
-                praat += ' --run'
-            }
         }
 
         project.task('templates', type: LegacyTemplateTask) {

--- a/legacy-plugin/src/main/resources/de/dfki/mary/voicebuilding/templates/database.config
+++ b/legacy-plugin/src/main/resources/de/dfki/mary/voicebuilding/templates/database.config
@@ -32,7 +32,7 @@ db.voicename ${project.voice.name}
 db.wavDir ${project.buildDir}/wav/
 
 # Properties for module PraatPitchmarker:
-PraatPitchmarker.command ${project.praat}
+PraatPitchmarker.command ${ def proc = 'praat --version'.execute(); proc.waitFor(); proc.exitValue() ? 'praat' : 'praat --run' }
 PraatPitchmarker.maxPitch ${project.voice.gender == 'female' ? 500 : 300}
 PraatPitchmarker.minPitch ${project.voice.gender == 'female' ? 100 : 75}
 PraatPitchmarker.pmDir ${project.buildDir}/pm/

--- a/legacy-plugin/src/main/resources/de/dfki/mary/voicebuilding/templates/database.config
+++ b/legacy-plugin/src/main/resources/de/dfki/mary/voicebuilding/templates/database.config
@@ -32,7 +32,7 @@ db.voicename ${project.voice.name}
 db.wavDir ${project.buildDir}/wav/
 
 # Properties for module PraatPitchmarker:
-PraatPitchmarker.command ${ def proc = 'praat --version'.execute(); proc.waitFor(); proc.exitValue() ? 'praat' : 'praat --run' }
+PraatPitchmarker.command ${ def proc = 'praat --version'.execute(); proc.waitForOrKill(100); proc.exitValue() ? 'praat' : 'praat --run' }
 PraatPitchmarker.maxPitch ${project.voice.gender == 'female' ? 500 : 300}
 PraatPitchmarker.minPitch ${project.voice.gender == 'female' ? 100 : 75}
 PraatPitchmarker.pmDir ${project.buildDir}/pm/

--- a/legacy-plugin/src/main/resources/de/dfki/mary/voicebuilding/templates/database.config
+++ b/legacy-plugin/src/main/resources/de/dfki/mary/voicebuilding/templates/database.config
@@ -32,7 +32,7 @@ db.voicename ${project.voice.name}
 db.wavDir ${project.buildDir}/wav/
 
 # Properties for module PraatPitchmarker:
-PraatPitchmarker.command ${ def proc = 'praat --version'.execute(); proc.waitForOrKill(100); proc.exitValue() ? 'praat' : 'praat --run' }
+PraatPitchmarker.command ${ def proc = 'praat --version'.execute(); proc.waitForOrKill(500); proc.exitValue() ? 'praat' : 'praat --run' }
 PraatPitchmarker.maxPitch ${project.voice.gender == 'female' ? 500 : 300}
 PraatPitchmarker.minPitch ${project.voice.gender == 'female' ? 100 : 75}
 PraatPitchmarker.pmDir ${project.buildDir}/pm/

